apps/condo/domains/onboarding/schema/UserHelpRequestFile.test.js

```diff
@@ -0,0 +1,258 @@
+                const [helpRequestFile] = await createTestUserHelpRequestFile(employeeUser)
+                const [updatedHelpRequestFile] = await updateTestUserHelpRequestFile(employeeUser, helpRequestFile.id, connectHelpRequestPayload)
+
+                expect(updatedHelpRequestFile.userHelpRequest.id).toEqual(employeeUserRequest.id)
+
+                const [otherHelpRequest] = await createTestUserHelpRequest(employeeUser, organization)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestUserHelpRequestFile(support, helpRequestFile.id, {
+                        userHelpRequest: { connect: { id: otherHelpRequest.id } },
+                    })
+                })
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestUserHelpRequestFile(support, helpRequestFile.id, {
+                        userHelpRequest: { disconnectAll: true },
+                    })
+                })
+            })
+
+            it('user with resident type: can not', async () => {
+                const [helpRequestFile] = await createTestUserHelpRequestFile(admin)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestUserHelpRequestFile(residentClient, helpRequestFile.id, connectHelpRequestPayload)
+                })
+            })
+        })
+
+        describe('Soft Delete', () => {
+            it('admin can', async () => {
+                const [helpRequestFile] = await createTestUserHelpRequestFile(admin)
+                const [deletedRequestFile] = await updateTestUserHelpRequestFile(admin, helpRequestFile.id, { deletedAt: 'true' })
+
+                expect(deletedRequestFile.deletedAt).toBeDefined()
+            })
+
+            it('anonymous can not', async () => {
+                const [helpRequestFile] = await createTestUserHelpRequestFile(admin)
+
+                await expectToThrowAuthenticationErrorToObj(async () => {
+                    await updateTestUserHelpRequestFile(anonymous, helpRequestFile.id, { deletedAt: 'true' })
+                })
+            })
+
+            it('support can not', async () => {
+                const [helpRequestFile] = await createTestUserHelpRequestFile(admin)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestUserHelpRequestFile(support, helpRequestFile.id, { deletedAt: 'true' })
+                })
+            })
+
+            it('user: can soft delete his help request file', async () => {
+                const [helpRequestFile] = await createTestUserHelpRequestFile(employeeUser)
+                const [deletedRequestFile] = await updateTestUserHelpRequestFile(employeeUser, helpRequestFile.id, { deletedAt: 'true' })
+
+                expect(deletedRequestFile.deletedAt).toBeDefined()
+            })
```

Comment:
What actually would happened on webhook side?

Reply:
Nothing
Webhook fires on `UserHelpRequest` model, not `UserHelpRequestFile`. And after user connect file to request, he is not able to delete this file (Connection occurs when user create `UserHelpRequest`)